packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingStringFilter.ts

```diff
@@ -21,7 +21,11 @@ export const isMatchingStringFilter = ({
       return regexCaseSensitive.test(value);
     }
     case stringFilter.ilike !== undefined: {
-      const regexPattern = stringFilter.ilike.replace(/%/g, '.*');
+      const escapedPattern = stringFilter.ilike.replace(
+        /[.*+?^${}()|[\]\\]/g,
+        '\\$&',
+      );
+      const regexPattern = escapedPattern.replace(/%/g, '.*');
```

Comment:
I think we can use [escapeRegExp](https://www.geeksforgeeks.org/lodash-_-escaperegexp-method/) from lodash instead of a custom one